iaso/models/tenant_users.py

```
@@ -1,5 +1,82 @@
+from dataclasses import dataclass
+
 from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
 from django.db import models
+from django.db.models import Q
+
+from iaso.models import Account
+
+
+@dataclass
+class UserCreationData:
+    account: Account
+    email: str
+    first_name: str
+    last_name: str
+    username: str
+
+
+class UsernameAlreadyExistsError(ValidationError):
+    pass
+
+
+def get_unique_username(username: str, account_name: str) -> str:
+    account_name_slug = account_name.lower().replace(" ", "_")
+    return f"{username}_{account_name_slug}"
```

In most cases I guess it will not be an issue, but there is a 150-char limit on usernames :warning:

---

I fixed this in https://github.com/BLSQ/iaso/pull/2246/commits/c32804b4160fdfa7f21bbbfb3f5f806750fef27d but I'm not 100% sure that this is the right call…

We may want that exact scenario to fail explicitly instead.